examples/resize/termsize.go

```diff
@@ -0,0 +1,39 @@
+// https://stackoverflow.com/a/15784352/587091
+
+package main
+
+import (
+	"runtime"
+	"syscall"
+	"unsafe"
+)
+
+const (
+	TIOCGWINSZ     = 0x5413
+	TIOCGWINSZ_OSX = 1074295912
+)
+
+type window struct {
+	Row    uint16
+	Col    uint16
+	Xpixel uint16
+	Ypixel uint16
+}
+
+func osTermSize() (int, int, error) {
+
+	w := new(window)
+	tio := TIOCGWINSZ
+	if runtime.GOOS == "darwin" {
+		tio = TIOCGWINSZ_OSX
+	}
+	res, _, err := syscall.Syscall(syscall.SYS_IOCTL,
+		uintptr(syscall.Stdin),
+		uintptr(tio),
+		uintptr(unsafe.Pointer(w)),
+	)
+	if int(res) == -1 {
+		return 0, 0, err
+	}
+	return int(w.Row), int(w.Col), nil
```

Comment:
Put newlines between blocks or grouped concepts (things that could be easily extracted into functions/methods. It make it a lot more readable, which is especially important for examples.

Reply:
Done